src/App.tsx

```diff
@@ -1,27 +1,65 @@
-import React from 'react';
+import React, { useState } from 'react';
+import Button from 'react-bootstrap/Button';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [allGoods, setAllGoods] = useState<Good[]>([]);
+  const [isSelected, setIsSelected] = useState(false);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  return (
+    <div className="App">
+      <h1>Dynamic list of Goods</h1>
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+      <Button
+        type="button"
+        data-cy="all-button"
+        variant="outline-primary"
+        className="Button"
+        onClick={() => {
+          getAll().then(goods => setAllGoods(goods));
+          setIsSelected(true);
+        }}
+      >
+        {allGoods.length === 0 && isSelected
+          ? 'Loading...'
+          : 'Load all goods'}
+      </Button>
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+      <Button
+        type="button"
+        data-cy="first-five-button"
+        variant="outline-warning"
+        className="Button"
+        onClick={() => {
+          get5First().then(goods => setAllGoods(goods));
+          setIsSelected(true);
+        }}
```

Comment:
look, you repeat yourself in all handlers, the only thing that changes is a method at a very start. Let's instead create universal handler that will accept all needed callback

Reply:
Also let's rewrite it using `async/await` syntax

Reply:
Also, [checklist](https://github.com/mate-academy/react_dynamic-list-of-goods/blob/master/checklist.md) 3rd item

Reply:
Also, everytime when I press the button, new request is created, even if this button was the last pressed one. 
![image](https://user-images.githubusercontent.com/37161315/228554406-67f79b5f-4759-44ee-9aa5-6717bf07881a.png)
Let's save last pressed button somehow and avoid reloading data if we press the same button again